apps/studio/src/components/Searchbar/SearchModal.tsx

```diff
@@ -0,0 +1,129 @@
+import { useState } from "react"
+import {
+  Modal,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  Text,
+} from "@chakra-ui/react"
+import { Searchbar as OgpSearchBar } from "@opengovsg/design-system-react"
+import { useDebounce } from "@uidotdev/usehooks"
+
+import type { SearchResultResource } from "~/server/modules/resource/resource.types"
+import { useBanner } from "~/hooks/useBanner"
+import { trpc } from "~/utils/trpc"
+import { isMac } from "./isMac"
+import {
+  InitialState,
+  LoadingState,
+  NoResultsState,
+  SearchResultsState,
+} from "./SearchModalBodyContentStates"
+
+interface SearchModalProps {
+  isOpen: boolean
+  onClose: () => void
+  siteId: string
+}
+export const SearchModal = ({ siteId, isOpen, onClose }: SearchModalProps) => {
+  const banner = useBanner()
+  const mt = banner ? "3rem" : "0.5rem"
+  const [searchValue, setSearchValue] = useState("")
+  const debouncedSearchTerm = useDebounce(searchValue, 300)
+  const { data, isLoading } = trpc.resource.search.useInfiniteQuery({
+    siteId,
+    query: debouncedSearchTerm,
+  })
+  const resources: SearchResultResource[] =
+    data?.pages.flatMap((page) => page.resources) ?? []
+
+  const renderModalBody = (): React.ReactNode => {
+    if (!!debouncedSearchTerm) {
+      if (isLoading) {
+        return <LoadingState />
+      }
+      if (resources.length === 0) {
+        return <NoResultsState />
+      }
+      return (
+        <SearchResultsState
+          siteId={siteId}
+          items={resources}
+          totalResultsCount={
+            data?.pages.reduce(
+              (acc, page) => acc + (page.totalCount ?? 0),
+              0,
+            ) ?? 0
+          }
+          searchTerm={debouncedSearchTerm}
+        />
+      )
+    }
+    return (
+      <InitialState
+        siteId={siteId}
+        items={data?.pages[0]?.suggestions.recentlyEdited ?? []}
+      />
+    )
+  }
+
+  return (
+    <Modal isOpen={isOpen} onClose={onClose} motionPreset="none">
+      <ModalOverlay bg="none" backdropFilter="brightness(80%)" />
+      <ModalContent
+        rounded="base"
+        w="42.5rem"
```

Comment:
i think will be pretty weird (and possibly unusable) if it's too wide. GDrive fixed it too

<img width="2546" alt="Screenshot 2024-11-22 at 5 11 38 PM" src="https://github.com/user-attachments/assets/0c69cb4f-c31d-45ee-a997-aeb4499b57d2">

Reply:
We can set a max width also so it doesn't get too wide. My concern is mainly because we want to support iPad editing so having this hardcoded makes it a little tougher

Reply:
both height and width fixed in https://github.com/opengovsg/isomer/pull/882/commits/15423aca484f5e31b3e3dddf33f94811acff034f

note - not optimized for anything smaller than an ipad mini

Reply:
https://github.com/user-attachments/assets/891d124d-fe7c-485b-9527-49f18941f155

Reply:
I think this looks good! Yeah anything smaller than iPad mini I think we don't need to bother for now unless design has special plans for those in the future.